gyp/pylib/gyp/generator/compile_commands_json.py

```diff
@@ -61,8 +61,9 @@ def AddCommandsForTarget(cwd, target, params, per_config_commands):
         defines = ["-D" + s for s in defines]
 
         # TODO(bnoordhuis) Handle generated source files.
+        extensions = (".c", ".cc", ".cpp", ".cxx")
         sources = target.get("sources", [])
-        sources = [s for s in sources if s.endswith(".c") or s.endswith(".cc")]
+        sources = [s for s in sources if s.endswith(extensions)]
```

Comment:
```suggestion
        sources = [s for s in target.get("sources", []) if s.endswith(extensions)]
```
And remove line 65 to streamline the logic.